### ISSUE TITLE
loop-var-prefix cannot be activated

### ISSUE THREAD

**Issue by niall-byrne** [ansible/ansible-lint#3564]
##### Summary

Galaxy runs ansible-lint 5.0.12 and finds violations of the `no-loop-var-prefix` rule.

Using main, (also tested 6.16, 6.17 series) I am unable to trigger the same rule.

Setup details:

Example Task File
```yaml
- name: Create Symlinks
  become: True
  become_user: "{{ symlinks_user }}"
  ansible.builtin.file:
    src: '{{ item.target }}'
    dest: '{{ item.link }}'
    state: link
  loop:
    "{{ symlinks }}"
```

Example Ansible-Lint Config:
```yaml
---
# .ansible-lint
loop_var_prefix: "^(__|{role}_)"
enable_list:
  - loop-var-prefix
```

I have tried disabling profiles as well, as part of the troubleshooting process:
```yaml
profile: null
```

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint 6.17.2.dev5 using ansible-core:2.15.0 ruamel-yaml:None ruamel-yaml-clib:None
```

- ansible installation method: pip and poetry have been tried
- ansible-lint installation method: pip and poetry have been tried

##### STEPS TO REPRODUCE

Create the playbook and config given above and run `ansible-lint`

```console (paste below)
ansible-lint tasks/main.yml
WARNING  Skipped installing old role dependencies due to running in offline mode.
WARNING  Skipped installing collection dependencies due to running in offline mode.

Passed: 0 failure(s), 0 warning(s) on 1 files. Last profile that met the validation criteria was 'production'.
You are using a pre-release version of ansible-lint.
```

##### Desired Behavior

I understand that this is supposed to fail under these settings.
- There could be a gap in my understanding of the configuration, or perhaps the behaviour of the rule is different than ansible-lint version 5?

##### Actual Behavior

ansible-lint is passing.  verbose output below.

```paste below
ansible-lint tasks/main.yml -vvv
DEBUG    Logging initialized to level 10
INFO     Identified /Volumes/Code/Code/osx-provisioner/role-symlinks as project root due config file /Volumes/Code/Code/osx-provisioner/role-symlinks/.ansible-lint.
DEBUG    Options: Options(cache_dir=PosixPath('/Users/nbyrne/.cache/ansible-compat/8bb37d'), colored=True, configured=True, cwd=PosixPath('/Volumes/Code/Code/osx-provisioner/role-symlinks'), display_relative_path=True, exclude_paths=['.cache', '.git', '.hg', '.svn', '.tox'], format=None, lintables=['tasks/main.yml'], list_rules=False, list_tags=False, write_list=[], parseable=False, quiet=False, rulesdirs=[PosixPath('/Volumes/Projects/Poetry/virtualenvs/role-symlinks-2VJSx_j1-py3.9/lib/python3.9/site-packages/ansiblelint/rules')], skip_list=[], tags=[], verbosity=3, warn_list=['experimental', 'jinja', 'fqcn'], mock_filters=[], mock_modules=[], mock_roles=[], loop_var_prefix=None, only_builtins_allow_collections=[], only_builtins_allow_modules=[], var_naming_pattern=None, offline=False, project_dir='.', extra_vars=None, enable_list=['loop-var-prefix'], skip_action_validation=True, strict=False, rules={}, profile=None, task_name_prefix='{stem} | ', sarif_file=None, config_file='/Volumes/Code/Code/osx-provisioner/role-symlinks/.ansible-lint', generate_ignore=False, rulesdir=[], cache_dir_lock=<filelock._unix.UnixFileLock object at 0x110b8cdf0>, use_default_rules=False, version=False, list_profiles=False, ignore_file=None)
DEBUG    CWD: /Volumes/Code/Code/osx-provisioner/role-symlinks
WARNING  Skipped installing old role dependencies due to running in offline mode.
WARNING  Skipped installing collection dependencies due to running in offline mode.
INFO     Set ANSIBLE_LIBRARY=/Users/nbyrne/.cache/ansible-compat/8bb37d/modules:/Users/nbyrne/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/Users/nbyrne/.cache/ansible-compat/8bb37d/collections:/Users/nbyrne/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/Users/nbyrne/.cache/ansible-compat/8bb37d/roles:/Users/nbyrne/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
DEBUG    Loading custom .yamllint.yml config file, this extends our internal yamllint config.
DEBUG    Effective yamllint rules used: {'anchors': {'level': 'error', 'forbid-undeclared-aliases': True, 'forbid-duplicated-anchors': False, 'forbid-unused-anchors': False}, 'braces': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 0, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'brackets': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 0, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'colons': {'level': 'error', 'max-spaces-before': 0, 'max-spaces-after': -1}, 'commas': {'level': 'error', 'max-spaces-before': 0, 'min-spaces-after': 1, 'max-spaces-after': -1}, 'comments': False, 'comments-indentation': False, 'document-end': False, 'document-start': {'level': 'error', 'present': True}, 'empty-lines': {'level': 'error', 'max': 3, 'max-start': 0, 'max-end': 0}, 'empty-values': False, 'float-values': False, 'hyphens': {'level': 'error', 'max-spaces-after': 1}, 'indentation': {'level': 'error', 'spaces': 'consistent', 'indent-sequences': True, 'check-multi-line-strings': False}, 'key-duplicates': {'level': 'error'}, 'key-ordering': False, 'line-length': False, 'new-line-at-end-of-file': {'level': 'error'}, 'new-lines': {'level': 'error', 'type': 'unix'}, 'octal-values': False, 'quoted-strings': False, 'trailing-spaces': {'level': 'error'}, 'truthy': False}
INFO     Running ansible-galaxy role install -vr requirements.yml --roles-path /Users/nbyrne/.cache/ansible-compat/8bb37d/roles
INFO     Running ansible-galaxy collection install -v -r requirements.yml
INFO     Set ANSIBLE_LIBRARY=/Users/nbyrne/.cache/ansible-compat/8bb37d/modules:/Users/nbyrne/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/Users/nbyrne/.cache/ansible-compat/8bb37d/collections:/Users/nbyrne/.cache/ansible-compat/8bb37d/collections:/Users/nbyrne/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/Users/nbyrne/.cache/ansible-compat/8bb37d/roles:/Users/nbyrne/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Using /Users/nbyrne/.cache/ansible-compat/8bb37d/roles/osx_provisioner.symlinks symlink to current repository in order to enable Ansible to find the role using its expected full name.
DEBUG    Examining tasks/main.yml of type tasks
DEBUG    Type checking ignored for 'force' option in task 'ansible.builtin.file' at line 4.
DEBUG    NAME 'symlinks_user' (prefix='')
DEBUG    NEWLINE '\n' (prefix=' ')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME 'symlinks_user' (prefix='')
DEBUG    NEWLINE '\n' (prefix='')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME 'symlinks' (prefix='')
DEBUG    NEWLINE '\n' (prefix=' ')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME 'symlinks' (prefix='')
DEBUG    NEWLINE '\n' (prefix='')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME 'item' (prefix='')
DEBUG    DOT '.' (prefix='')
DEBUG    NAME 'target' (prefix='')
DEBUG    NEWLINE '\n' (prefix=' ')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME 'item' (prefix='')
DEBUG    DOT '.' (prefix='')
DEBUG    NAME 'target' (prefix='')
DEBUG    NEWLINE '\n' (prefix='')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME 'item' (prefix='')
DEBUG    DOT '.' (prefix='')
DEBUG    NAME 'link' (prefix='')
DEBUG    NEWLINE '\n' (prefix=' ')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME 'item' (prefix='')
DEBUG    DOT '.' (prefix='')
DEBUG    NAME 'link' (prefix='')
DEBUG    NEWLINE '\n' (prefix='')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME 'false' (prefix='')
DEBUG    NAME 'if' (prefix=' ')
DEBUG    STRING '"force"' (prefix=' ')
DEBUG    NAME 'not' (prefix=' ')
DEBUG    NAME 'in' (prefix=' ')
DEBUG    NAME 'item' (prefix=' ')
DEBUG    NAME 'else' (prefix=' ')
DEBUG    NAME 'item' (prefix=' ')
DEBUG    DOT '.' (prefix='')
DEBUG    NAME 'force' (prefix='')
DEBUG    NEWLINE '\n' (prefix=' ')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME 'false' (prefix='')
DEBUG    NAME 'if' (prefix=' ')
DEBUG    STRING '"force"' (prefix=' ')
DEBUG    NAME 'not' (prefix=' ')
DEBUG    NAME 'in' (prefix=' ')
DEBUG    NAME 'item' (prefix=' ')
DEBUG    NAME 'else' (prefix=' ')
DEBUG    NAME 'item' (prefix=' ')
DEBUG    DOT '.' (prefix='')
DEBUG    NAME 'force' (prefix='')
DEBUG    NEWLINE '\n' (prefix='')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME 'true' (prefix='')
DEBUG    NAME 'if' (prefix=' ')
DEBUG    STRING '"force"' (prefix=' ')
DEBUG    NAME 'not' (prefix=' ')
DEBUG    NAME 'in' (prefix=' ')
DEBUG    NAME 'item' (prefix=' ')
DEBUG    NAME 'else' (prefix=' ')
DEBUG    NAME 'not' (prefix=' ')
DEBUG    NAME 'item' (prefix=' ')
DEBUG    DOT '.' (prefix='')
DEBUG    NAME 'force' (prefix='')
DEBUG    NEWLINE '\n' (prefix=' ')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    NAME 'true' (prefix='')
DEBUG    NAME 'if' (prefix=' ')
DEBUG    STRING '"force"' (prefix=' ')
DEBUG    NAME 'not' (prefix=' ')
DEBUG    NAME 'in' (prefix=' ')
DEBUG    NAME 'item' (prefix=' ')
DEBUG    NAME 'else' (prefix=' ')
DEBUG    NAME 'not' (prefix=' ')
DEBUG    NAME 'item' (prefix=' ')
DEBUG    DOT '.' (prefix='')
DEBUG    NAME 'force' (prefix='')
DEBUG    NEWLINE '\n' (prefix='')
DEBUG    ENDMARKER '' (prefix='')
DEBUG    Stop.
DEBUG    Attempting to release lock 4575514096 on /Users/nbyrne/.cache/ansible-compat/8bb37d/.lock
DEBUG    Lock 4575514096 released on /Users/nbyrne/.cache/ansible-compat/8bb37d/.lock

DEBUG    Determined rule-profile order: {'internal-error': (0, 'min'), 'load-failure': (1, 'min'), 'parser-error': (2, 'min'), 'syntax-check': (3, 'min'), 'command-instead-of-module': (4, 'basic'), 'command-instead-of-shell': (5, 'basic'), 'deprecated-bare-vars': (6, 'basic'), 'deprecated-local-action': (7, 'basic'), 'deprecated-module': (8, 'basic'), 'inline-env-var': (9, 'basic'), 'key-order': (10, 'basic'), 'literal-compare': (11, 'basic'), 'jinja': (12, 'basic'), 'no-free-form': (13, 'basic'), 'no-jinja-when': (14, 'basic'), 'no-tabs': (15, 'basic'), 'partial-become': (16, 'basic'), 'playbook-extension': (17, 'basic'), 'role-name': (18, 'basic'), 'schema': (19, 'basic'), 'name': (20, 'basic'), 'var-naming': (21, 'basic'), 'yaml': (22, 'basic'), 'name': (23, 'moderate'), 'name': (24, 'moderate'), 'name': (25, 'moderate'), 'spell-var-name': (26, 'moderate'), 'avoid-implicit': (27, 'safety'), 'latest': (28, 'safety'), 'package-latest': (29, 'safety'), 'risky-file-permissions': (30, 'safety'), 'risky-octal': (31, 'safety'), 'risky-shell-pipe': (32, 'safety'), 'galaxy': (33, 'shared'), 'ignore-errors': (34, 'shared'), 'layout': (35, 'shared'), 'meta-incorrect': (36, 'shared'), 'meta-no-tags': (37, 'shared'), 'meta-video-links': (38, 'shared'), 'meta-version': (39, 'shared'), 'meta-runtime': (40, 'shared'), 'no-changed-when': (41, 'shared'), 'no-changelog': (42, 'shared'), 'no-handler': (43, 'shared'), 'no-relative-paths': (44, 'shared'), 'max-block-depth': (45, 'shared'), 'max-tasks': (46, 'shared'), 'unsafe-loop': (47, 'shared'), 'avoid-dot-notation': (48, 'production'), 'sanity': (49, 'production'), 'fqcn': (50, 'production'), 'import-task-no-when': (51, 'production'), 'meta-no-dependencies': (52, 'production'), 'single-entry-point': (53, 'production'), 'use-loop': (54, 'production')}
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
I am almost sure that this is caused by the fact that you feed the file path to the linter. For this rule to work it needs to detect that this file is part of a role, basically a patch like `roles/foo/tasks/main.yml`, note that without having roles before it would not count as a role, as 'foo' could be a random folder as well.
